### ISSUE TITLE
[Question] Variable/dropdown Block value

### ISSUE THREAD

**Issue by theoriginalbit** [espertus/blockly-lua#7]
Is it possible for a block to have a drop down, that lets say says
- all
- other

and when `other` is selected they can drag a variable and connected it to the block, but when `all` is selected the default value is used...

I think if this is possible it could make turtle.drop, turtle.refuel, and turtle.transferTo a lot simpler to use for the newbies.

**Comment by espertus** [comment]
That's a good idea.  I'd have to think about it.  One easy workaround would be to create two blocks, as I did with place and place_sign, so we would have `turtle.drop_all` and `turtle_drop`, whose text would be "drop up to __ items".

**Comment by theoriginalbit** [comment]
yeh I originally had the same thought of defining new blocks, but thought that this would be better if it was already implemented, hence the question... For the interim I'll define some extra blocks though :)

**Comment by espertus** [comment]
I can implement either of these:
1. Create a block with a dropdown with choices "all" (initially selected) or "some".  If the user switches to "some", the dropdown gets replaced by a socket for a block.  Problem: There'd be no way of converting back to the dropdown having "all".  OTOH, they could always drag out a new block from the toolbar.
2. Use a context menu to switch modes.  I'm afraid this might be hard for users to find.

What do you think?

**Comment by espertus** [comment]
For grins, I implemented the GUI for turtle_drop for the first option on a test server [http://test.blockly-lua.appspot.com/static/apps/code/index.html].  Could you let me know what you think?

**Comment by theoriginalbit** [comment]
Yep love it. Exactly what I was thinking when I original thought of this suggestion, makes blocks just that little more powerful.

One suggestion, which is a change in wording and allows for the drop down to stick around meaning they can change it back if needed...

"drop `all` items from the selected slot `DIR`"
and
"drop `up to` `|` items from the selected slot `DIR`"

where `|` is the block connector

Also means that now you could have

"place `block` from the selected slot `DIR`"
and
"place `sign` from the selected slot `DIR` with text `|`"

**Comment by espertus** [comment]
Good suggestions on wording.  Should "block" be "item"?

I agree it's better to have fewer, more powerful, blocks.

**Comment by theoriginalbit** [comment]
block/item maybe, idk I'll leave that decision up to you ;)

EDIT: oh btw links like this [Google](www.google.com) are [name].(link) without the '.'

**Comment by theoriginalbit** [comment]
Well see technically if the turtle is placing cobble stone or the like, I'd put block (even though in code its called an ItemBlock) but then if its placing like redstone, I'd put item.... So maybe just put `block/item`?

**Comment by espertus** [comment]
We could say "contents of", although that's not ideal either.  I'll just do
"item" for now.

Take a look at some changes I made to the block:
http://test.blockly-lua.appspot.com/static/apps/code/index.html

Specifically, what do you think of the change between "some" and "up to"?
 Note that changing back is now supported.

Ellen

**Comment by espertus** [comment]
I added  an "all" block, currently plugged into Turtle.refuel.  Could you
take a look at it at the test server [
http://test.blockly-lua.appspot.com/static/apps/code/index.html]?  It is
much simpler to implement than the fancy mutation I have for "drop", so I'm
inclined to switch to using an "all" block.  What do you think?

**Comment by theoriginalbit** [comment]
I think it either needs to stick with "some" or "up to" the switching between is a little odd. Other than that its perfect, especially if its easier to implement ;)
